agreement_account/models/agreement.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# © 2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models, fields
+
+
+class Agreement(models.Model):
+    _name = 'agreement'
+    _description = 'Agreement'
+
+    code = fields.Char(
+        string='Code', required=True, copy=False)
+    name = fields.Char(string='Name', required=True)
+    agreement_type = fields.Selection([
+        ('sale', 'Sale'),
+        ('purchase', 'Purchase'),
+        ], string='Type', required=True)
+    partner_id = fields.Many2one(
+        'res.partner', string='Partner', ondelete='restrict', required=True,
+        domain=[('parent_id', '=', False)])
+    company_id = fields.Many2one(
+        'res.company', string='Company',
+        default=lambda self: self.env['res.company']._company_default_get(
+            'agreement'))
+    active = fields.Boolean(string='Active', default=True)
+    signature_date = fields.Date(string='Signature Date')
+    out_invoice_ids = fields.One2many(
+        'account.invoice', 'agreement_id', string='Customer Invoices',
+        readonly=True, domain=[('type', 'in', ('out_invoice', 'out_refund'))])
+    in_invoice_ids = fields.One2many(
+        'account.invoice', 'agreement_id', string='Supplier Invoices',
+        readonly=True, domain=[('type', 'in', ('in_invoice', 'in_refund'))])
+
+    def name_get(self):
```

Comment:
Must decorate api.multi

Reply:
As far as I know all methods which inherit of models.Model are `api.multi` by default in v10 (dixit Antony Lesuisse @ last odoo days). Then we should only specify decorator when don't want this behavior.
It allows to make the code lighter. 
Sure it was different in previous versions.